tests/shared/src/test/scala/org/http4s/DecodeSpec.scala

```diff
@@ -81,44 +81,24 @@ class DecodeSpec extends Http4sSuite {
       val expected = new String(source.toVector.toArray, cs.nioCharset)
       !expected.contains("\ufffd") ==> {
         // \ufffd means we generated a String unrepresentable by the charset
-        val decoded = source.through(decode[Fallible](cs)).compile.string
+        val decoded = source.through(decodeWithCharset[Fallible](cs.nioCharset)).compile.string
         decoded == Right(expected)
       }
     }
   }
 
   test("decode should drop Byte Order Mark") {
     val source = Stream(0xef.toByte, 0xbb.toByte, 0xbf.toByte)
-    val decoded = source.through(decode[Fallible](Charset.`UTF-8`)).compile.string
+    val decoded = source.through(decodeWithCharset[Fallible](StandardCharsets.UTF_8)).compile.string
     decoded == Right("")
   }
 
-  test("decode should handle malformed input") {
-    // Not a valid first byte in UTF-8
-    val source = Stream(0x80.toByte)
-    val decoded = source.through(decode[Fallible](Charset.`UTF-8`)).compile.string
-    assert(decoded match {
-      case Left(_: MalformedInputException) => true
-      case _ => false
-    })
-  }
-
-  test("decode should handle incomplete input") {
-    // Only the first byte of a two-byte UTF-8 sequence
-    val source = Stream(0xc2.toByte)
-    val decoded = source.through(decode[Fallible](Charset.`UTF-8`)).compile.string
-    assert(decoded match {
-      case Left(_: MalformedInputException) => true
-      case _ => false
-    })
-  }
-
```

Comment:
I assume this is tested upstream?

Reply:
We never called the deprecated `decode` publicly on UTF-8.  We optimize to `fs2.text.utf8.decode`. Unfortunately, that has different error semantics than our `decode` and the new FS2 `decodeWithCharset`.  That's another one of those semantic changes we need to wrestle with.

Reply:
I'll open an issue in FS2, because our old inconsistent behavior is now also their inconsistent behavior.